autowarefoundation/autoware
# Removing CUDA dependencies for Darknet Yolov3

## Status
**PRODUCTION / DEVELOPMENT**

## Description
Fix autowarefoundation/autoware_ai#455 
Fix autowarefoundation/autoware#1782 due to failed TravisCI checks 

Moving to develop branch instead of master

If the host machine does not have CUDA, this will attempt to build the vision_darknet_detect package based on a pre-built darknet directory (which doesn't require CUDA as there are no CUDA dependencies for yolov3).

## Steps to Test or Reproduce
If the system has CUDA there should be no issue. If CUDA is not present then darknet must be build in the home directory:
```
$ cd && git clone https://github.com/pjreddie/darknet.git
$ cd ~/darknet && make
```
After darknet has been built, the Autoware node for vision_darknet_detect can be recompiled, and should have no issues:
In order to just compile vision_darknet_detect the following command can be executed:
```
$ catkin_make -DCATKIN_WHITELIST_PACKAGES="vision_darknet_detect"
```
Now the following should run with no errors on a system that does not have CUDA
```
roslaunch vision_darknet_detect vision_darknet_detect
```
